docs/DeveloperGuide.md

```diff
@@ -1,38 +1,212 @@
+- `help` : Shows the help page.  
+- `bye` : Exits the program.  
+
+```
+____________________________________________________________
+____    __    ____  __   __       _______
+\   \  /  \  /   / |  | |  |     |       \
+ \   \/    \/   /  |  | |  |     |  .--.  |
+  \            /   |  | |  |     |  |  |  |
+   \    /\    /    |  | |  `----.|  '--'  |
+    \__/  \__/     |__| |_______||_______/
+____    __    ____  ___   .___________.  ______  __    __
+\   \  /  \  /   / /   \  |           | /      ||  |  |  |
+ \   \/    \/   / /  ^  \ `---|  |----`|  ,----'|  |__|  |
+  \            / /  /_\  \    |  |     |  |     |   __   |
+   \    /\    / /  _____  \   |  |     |  `----.|  |  |  |
+    \__/  \__/ /__/     \__\  |__|      \______||__|  |__|
+____________________________________________________________
+Hello there! Welcome to WildWatch!
+
+Checking if "WildWatch.txt" already exists...
+File does not exist.
+Creating new file...
+File created successfully.
+What would you like to do?
+____________________________________________________________
+>>> 
+```
+
+[⬆ Back to top](https://ay2324s1-cs2113t-w11-2.github.io/tp/DeveloperGuide.html#table-of-contents)
+
+--------------------------------------------------------------------------------------------------------------------------------------
 
-## Acknowledgements
+## Design & Implementation
 
-{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
+### System Architecture
 
-## Design & implementation
+![System Architecture](https://github.com/AY2324S1-CS2113T-W11-2/tp/assets/69474977/7bdda637-1c7a-4358-a648-a9972184cd3d)
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+The **_Architecture Diagram_** shows the high-level architectural design of WildWatch.  
+Actual detailed implementation may differ from the architectural diagram.  
+WildWatch is comprised of 7 major components.  
+- `Main`: Entry point of the program.
+- `UI`: A class that receives input from, and prints output to the user.
+- `Parser`: A class that processes and interpretes the input command of the user.
+- `Command`: A set of classes that does specific task according to the command of the user.
+- `EntryList`: A class that stores all the wildlife data, while the program is running.
+- `Entry`: A class that stores individual wildlife data, while the program is running.
+- `FileHandler`: A class that saves and retrieves all the wildlife data from the local storage
 
+### Parser component 
+The `Parser` component is responsible for for parsing an input from the user, and returns an `XYZCommand` class. 
+
+### Command component 
+//TODO: incomplete
+For the diagram below, the command `add D/02-03-23 S/Annam Leaf Turtle N/Ariel R/Injured left flipper` has been substituted by `input`. 
+
+![](diagrams/AddSequenceDiagram.svg)
+
```

Comment:
AddCommand should be destroyed after being executed and returns a command result.